src/main/java/net/minecraftforge/client/model/ItemLayerModel.java

```diff
@@ -5,488 +5,170 @@
-import net.minecraft.client.renderer.texture.TextureAtlasSprite;
-import com.mojang.blaze3d.vertex.VertexFormat;
-import com.mojang.blaze3d.vertex.VertexFormatElement;
-import net.minecraft.server.packs.resources.ResourceManager;
-import net.minecraft.core.Direction;
-import net.minecraft.util.GsonHelper;
-import net.minecraft.resources.ResourceLocation;
-import com.mojang.math.Transformation;
-import net.minecraftforge.client.ForgeRenderTypes;
-import net.minecraftforge.client.model.geometry.IModelGeometry;
-import net.minecraftforge.client.model.pipeline.BakedQuadBuilder;
-import net.minecraftforge.client.model.pipeline.IVertexConsumer;
-import net.minecraftforge.client.model.pipeline.TRSRTransformer;
-
-import java.util.*;
-import java.util.function.Function;
-
-import net.minecraft.client.renderer.block.model.BakedQuad;
 import net.minecraft.client.renderer.block.model.ItemOverrides;
-import net.minecraft.client.renderer.block.model.ItemTransforms;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
 import net.minecraft.client.resources.model.BakedModel;
 import net.minecraft.client.resources.model.Material;
 import net.minecraft.client.resources.model.ModelBakery;
 import net.minecraft.client.resources.model.ModelState;
 import net.minecraft.client.resources.model.UnbakedModel;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraftforge.client.ForgeRenderTypes;
+import net.minecraftforge.client.RenderTypeGroup;
+import net.minecraftforge.client.model.geometry.IGeometryBakingContext;
+import net.minecraftforge.client.model.geometry.IGeometryLoader;
+import net.minecraftforge.client.model.geometry.IUnbakedGeometry;
+import net.minecraftforge.client.model.geometry.UnbakedGeometryHelper;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.jetbrains.annotations.Nullable;
 
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+
 /**
- * Forge reimplementation of vanilla {@link ItemModelGenerator}, i.e. builtin/generated models,
- * with the following changes:
- * - Represented as a true {@link UnbakedModel} so it can be baked as usual instead of using
- *   special-case logic like vanilla does.
- * - Various fixes in the baking logic.
- * - Not limited to 4 layers maximum.
+ * Forge reimplementation of vanilla's {@link ItemModelGenerator}, i.e. builtin/generated models with some tweaks:
+ * - Represented as {@link IUnbakedGeometry} so it can be baked as usual instead of being special-cased
+ * - Not limited to an arbitrary number of layers (5)
+ * - Support for per-layer render types
  */
-public final class ItemLayerModel implements IModelGeometry<ItemLayerModel>
+public class ItemLayerModel implements IUnbakedGeometry<ItemLayerModel>
 {
-    public static final ItemLayerModel INSTANCE = new ItemLayerModel(ImmutableList.of());
+    private static final Logger LOGGER = LogManager.getLogger();
```

Comment:
As discussed, since loggers in this PR were borrowed from other Forge code and seems like a lot of places are still using the Log4J API instead of SLF4J, I'll keep it as-is and it can be updated in a future cleanup pass if Forge decides to fully transition to native SLF4J.